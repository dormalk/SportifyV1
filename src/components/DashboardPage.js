import React from 'react';
import { connect } from 'react-redux';
import {startLogout} from '../actions/auth';

export const DashboardPage = ({startLogout}) => (
    <div>
        <p>wellcome my frined</p>
        <button onClick={startLogout}>Logout</button>
    </div>
);


const mapPropsToDispatch = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});


export default connect(undefined,mapPropsToDispatch)(DashboardPage);
