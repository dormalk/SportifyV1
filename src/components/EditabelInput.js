import React from 'react';
//Test
export default class EditabelInput extends React.Component{
    constructor(props){
        super();
        this.state = {
            disable: true
        }
    }

    onEditClick = () => {
        const disable = !this.state.disable;
        this.setState({disable});
    }

    render(){
        return(
            <div>
                <label><span className="input_label">{this.props.label}</span>
                    <input type="text" value={this.props.val} onChange={this.props.onChangeValue} disabled={this.state.disable}/>
                    {this.state.disable? (
                        <span className="start__edit_input" onClick={this.onEditClick}>ערוך/י</span>
                    ):(
                        <span  className="end__edit_input" onClick={this.onEditClick}>סיימ/י עריכה</span>           
                    )}
                </label>
            </div>
        )        
    }
}
