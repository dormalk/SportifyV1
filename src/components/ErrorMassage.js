import React from 'react';


export default (props) => (
    <div>
    שגיאה התרחשה - {props.error}
    </div>
) 
