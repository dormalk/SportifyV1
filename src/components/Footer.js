import React from 'react';


export default () => (
    <div className="show-for-desktop footer">
        <span>כל הזכויות שמורות</span>
    </div>
)