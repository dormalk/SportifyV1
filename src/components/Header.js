import React from 'react';
import { connect } from 'react-redux';

export const Header = () => (
    <div className="header">
        <div className="content-container">
            <h1 className="header__title">Sportify</h1>
        </div>
    </div>
);

export default Header;
