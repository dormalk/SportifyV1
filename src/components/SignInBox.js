import React from 'react';
import { connect } from 'react-redux';
import validator from 'validator';
import { Link } from 'react-router-dom';
import { startLoginOnGoogle } from '../actions/auth';
import { startLoginOnEmail } from '../actions/auth';

export class SignInBox extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            email: this.props.email || '', 
            password: this.props.password || '',
            error: this.props.error || ''
        }
    };

    onEmailInputChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    };

    onPasswordInputChange = (e) => {
        const password = e.target.value;
        if(password.length < 20){
            this.setState(() => ({ password }));
        }
    };

    onSubmit = (e) => {
        e.preventDefault();
        if(!validator.isEmail(this.state.email)){
            this.setState(() => ({ error:'Not valid email' }));            
        }
        else if(this.state.password.length < 6) {
            this.setState(() => ({ error:'Not valid password' }));
        }
        else{
            this.setState(() => ({ error:'' }));
            this.props.startLoginOnEmail(this.state.email,this.state.password);
        }
    }

    render() {
        return (
            <div className="content-container">
                <div className="login-box">
                    <h1 className="login-box__title">Login Sportify</h1>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}
                    <form  className="login-form" onSubmit={this.onSubmit}>
                        <input 
                            type="text"
                            placeholder="Email"
                            autoFocus
                            value={this.state.email}
                            onChange={this.onEmailInputChange}  
                        />
                        <input 
                            type="password"
                            placeholder="Password"
                            value={this.state.password}
                            onChange={this.onPasswordInputChange}  
                        />
                        <div className="login__non-provider">
                            <button type="submit" className="login_email">SIGN IN</button>
                            <Link className="signup" to="/signup">CREATE ACCOUNT</Link>
                        </div>
                        <button type="button" className="login__google" onClick={this.props.startLoginOnGoogle}>SIGN IN WITH GOOGLE</button>
                    </form>
                </div>
            </div>
        );
    };
}; 



const mapPropsToDispatch = (dispatch) => ({
    startLoginOnGoogle: () => dispatch(startLoginOnGoogle()),
    startLoginOnEmail: (email,password) => dispatch(startLoginOnEmail(email,password)),
});


export default connect(undefined,mapPropsToDispatch)(SignInBox);