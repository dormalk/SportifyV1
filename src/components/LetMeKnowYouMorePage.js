//Test
import React from 'react';
import { connect } from 'react-redux';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import {isPartOfHobbie,isHobbie,fields} from '../selectors/HobbiesSuggestion'; 
import EditabelInput from './EditabelInput';


export const CheckDetail = ({fname,lname,onChangeFname,onChangeLname,onContinueButton}) => (
    <div className="check_detail">
        <h2>האם הפרטים נכונים?</h2>
        <EditabelInput val={fname} onChangeValue={onChangeFname} label={"שם פרטי"}/>
        <EditabelInput val={lname} onChangeValue={onChangeLname} label={"שם משפחה"}/>
        <button className="confirm_detail" onClick={onContinueButton}>אישור</button>
    </div>
)

export const YourGender = ({hendelGenderChange}) => {
    const genders = ['זכר','נקבה','לא מעוניין למסור']
    return(
        <div className="your_gender">
            <h2>מהו מינך?</h2>
            {genders.map((gender,i) => (
                <label key={i}>
                    <input 
                        key={i}
                        onChange={hendelGenderChange}
                        type='radio'
                        name='gender'
                        value={gender}/>
                        <span className="radiomark"><span className="radiomark_checked"></span></span>
                        {gender}
                </label>
            ))}
    </div>
    )
}

export const YourAge = ({hendelAgeChange}) => (
    <div>
        <h2>גיל</h2>
        <input type="number" placeholder="18" min="16" max="99" onChange={hendelAgeChange}/>
    </div>
);

export const YourHobbies = ({currHobbie,Hobbies,onHobbieEntered,onHobbieRemoved,onHobbieChanged,onHobbieChangedFromSuggest,suggestHobbie}) => (
    <div>
        <h2>מה התחביבים שלך?</h2>
        <input type="text" value={currHobbie} onChange={onHobbieChanged.bind(this)} onKeyPress={onHobbieEntered.bind(this)}/>
        {Hobbies.map((hobbie,i) => (<div key={i}>{hobbie}<button onClick={onHobbieRemoved.bind(this,hobbie)}>x</button></div>))}
        {currHobbie!='' && suggestHobbie(currHobbie).map((val,i) => (<div key={i} onClick={onHobbieChangedFromSuggest.bind(this,val)}>{val}</div>))}
    </div>
)

export class LetMeKnowYouMorePage extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            user: {
                ...this.props.user,
                detail: {
                    gender: '',
                    age:'',
                    hobbies: []    
                }
            },
            autocompleat:'',
            flow: 1,
        }
        
        this.isStateHobbie = this.isStateHobbie.bind(this);
    }
    
    onChangeFname = (e) => {
        const fname = e.target.value;
        const user = {
            ...this.state.user,
            fname,
            detail: {
               ...this.state.user.detail 
            }
        }
        this.setState({user});
    };


    onChangeLname = (e) => {
        const lname = e.target.value;
        const user = {
            ...this.state.user,
            lname,
            detail: {
               ...this.state.user.detail 
            }
        }
        this.setState({user});
    };


    hendelGenderChange = (e) => {
        const gender = e.target.value;
        const user = {
            ...this.state.user,
            detail: {
               ...this.state.user.detail,
               gender 
            }
        }
        this.setState({user});
    }

    hendelAgeChange = (e) => {
        const age = e.target.value;
        const user = {
            ...this.state.user,
            detail: {
               ...this.state.user.detail,
               age
            }
        }
        this.setState({user});
    }

    isStateHobbie = (value) => {
        var _isHobbie = true;
        const _fields = this.state.user.detail.hobbies;
        for(var i=0;i<_fields.length;i++)
            if(_fields[i] === value){
                _isHobbie = false;
                break;
            } 
        return _isHobbie;    
    }

    onHobbieEntered = (e) => {
        const hobbie = e.target.value;
        if(e.key === 'Enter'){
            if(isHobbie(hobbie) && this.isStateHobbie(hobbie)){
                const autocompleat = '';
                const hobbies = this.state.user.detail.hobbies;
                hobbies.push(hobbie);    
                const user = {
                    ...this.state.user,
                    detail: {
                       ...this.state.user.detail,
                       hobbies
                    }
                }        
                this.setState({user});
                this.setState({autocompleat})    
            }
        }
    }

    onHobbieRemoved = (hobbie) => {
        var hobbies = this.state.user.detail.hobbies.filter((h) => {return h !== hobbie;});
        const user = {
            ...this.state.user,
            detail: {
               ...this.state.user.detail,
               hobbies
            }
        }        
        this.setState({user});
    }

    onHobbieChanged = (e) => {
        const autocompleat = e.target.value;
        if(isPartOfHobbie(autocompleat))
            this.setState({autocompleat});
    }

    suggestHobbie = (value) => {
        return fields.filter((v) => {return v.startsWith(value) && this.isStateHobbie(v)});
    }

    onHobbieChangedFromSuggest = (value) => {
        const hobbie = value;
        const hobbies = this.state.user.detail.hobbies;
        hobbies.push(hobbie);
        const autocompleat = '';
        const user = {
            ...this.state.user,
            detail: {
               ...this.state.user.detail,
               hobbies
            }
        }        
        this.setState({user});
        this.setState({autocompleat});
    }


    onContinueButton = () => {
        this.setState({flow: this.state.flow+1});
    }

    render(){
        return this.state.user &&
        (
            <div className="content-container">
                <div className="let_me_know_you_more">
                    {this.state.user.fname && <h1>היי  {this.state.user.fname}, אני רוצה להכיר אותך יותר!</h1>}
                    <ReactCSSTransitionGroup
                        transitionName="exm"
                        transitionAppear={false}
                        transitionEnterTimeout={500}
                        transitionEnter={true}
                        transitionLeave={false}>
                        {
                            this.state.flow === 1 && 
                            <div>
                                <CheckDetail 
                                    fname={this.state.user.fname} 
                                    lname={this.state.user.lname}
                                    onChangeFname={this.onChangeFname}
                                    onChangeLname={this.onChangeLname}
                                    onContinueButton={this.onContinueButton}/>
                            </div>
                        }
                        {this.state.flow === 2 && <div className="grey_border_box_display">
                            <ReactCSSTransitionGroup
                            transitionName="exm"
                            transitionAppear={false}
                            transitionEnterTimeout={500}
                            transitionEnter={true}
                            transitionLeave={false}>
                                <YourGender hendelGenderChange={this.hendelGenderChange}/>
                                {this.state.user.detail.gender && <YourAge hendelAgeChange={this.hendelAgeChange}/>}
                                {this.state.user.detail.age && <YourHobbies 
                                                        Hobbies={this.state.user.detail.hobbies} 
                                                        onHobbieEntered={this.onHobbieEntered}
                                                        onHobbieRemoved={this.onHobbieRemoved}
                                                        onHobbieChanged={this.onHobbieChanged}
                                                        currHobbie={this.state.autocompleat}
                                                        onHobbieChangedFromSuggest={this.onHobbieChangedFromSuggest}
                                                        suggestHobbie={this.suggestHobbie}/>}
                                {this.state.user.detail.hobbies.length > 0 &&
                                <button>זהו סיימתי</button>}
                            </ReactCSSTransitionGroup>
                        </div>}
                    </ReactCSSTransitionGroup>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    user: state.user
});

export default connect(mapStateToProps,undefined)(LetMeKnowYouMorePage);
