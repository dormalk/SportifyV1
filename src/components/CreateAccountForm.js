import React from 'react';
import { connect } from 'react-redux';
import { startUploadImage } from '../actions/storage';
import { startCreateNewUser } from '../actions/user';
import ErrorMassage from './ErrorMassage';


export class CreateAccountForm extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            user:{
                fname: this.props.fname || '',
                lname: this.props.lname || '',
                email: this.props.email || '',
                password: this.props.password || '',
                profile: this.props.profile || ''       
            },
            c_password: '',
            error: ''
        }
    }

    onSubmit = (e) => {
        e.preventDefault();
        this.props.startCreateNewUser(this.state.user).then(() => {
            this.setState(() => ({ error:'' }));
        }).catch((error) => {
            if(error.code === 'auth/email-already-in-use'){
                this.setState(() => ({ error:'המייל שהזנת נימצא בשימוש' }));
            }
            
        });
    };

    onFnameChange = (e) => {
        const fname = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, fname } } ));
    }

    onLnameChange = (e) => {
        const lname = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, lname } } ));
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ( { user: { ...this.state.user, email } } ));
    }

    onPasswordChange = (e) => {
        const password = e.target.value;
        if(password.length < 20) {
            this.setState(() => ( { user: { ...this.state.user, password } } ));
        }
    }

    onConfirmPasswordChange = (e) => {
        const c_password = e.target.value;
        if(c_password.length < 20) {
            this.setState(() => ( { c_password } ));
        }
    }

    checkCompatibilityPasswords = (e) => {
        const r_password = this.state.user.password;
        const c_password = this.state.c_password;
        if(c_password !== r_password){
            this.setState(() => ({ error:'Passwords do not match' }));
        }
        else{
            this.setState(() => ({ error:'' }));
        }
    }

    onProfileChange = (e) => {
        const picture = e.target.files[0];
        if(picture.type !== 'image/jpeg' && picture.type !== 'image/jpg' && picture.type !== 'image/png'){
            this.setState({error:'נא בחר קובץ תמונה (jpg,jpeg,png)'});
        }
        else if(picture.size > 585750){
            this.setState({error:'נא בחר קובץ תמונה פחות מ 585,750 בית'});
            console.log('Hello');
        }
        else{
                this.setState(() => ({error:''}));  
                this.props.startUploadImage(picture).then((profile) => {
                    this.setState({error:''});
                    this.setState(() => ( { user: { ...this.state.user, profile } } ));
                }).catch((error) => {
                    this.setState({error:'לא ניתן לעלות קובץ תמונה ברגע זה'});
                });
        }
    }

    render(){
        return (
            <div>
                {this.state.error && <ErrorMassage error={this.state.error}/>}
                <div className="content-container">
                    <div className="signup-box">
                        <form 
                            onSubmit={this.onSubmit}
                            className="signup-form"
                        >
                            <h2>יצירת משתמש חדש</h2>
                            {this.state.error && <p className="form__error">{this.state.error}</p>}
                            <label>שם פרטי
                                <input 
                                    type="text"
                                    onChange={this.onFnameChange}
                                    required
                                />
                            </label>
                            <label>שם משפחה
                                <input 
                                    type="text"
                                    onChange={this.onLnameChange}
                                    required
                                />
                            </label>
                            <label>אימייל
                                <input 
                                    type="text"
                                    onChange={this.onEmailChange}
                                    required
                                />
                            </label>
                            <label>סיסמא
                                <input 
                                    type="password"
                                    onChange={this.onPasswordChange}
                                    required
                                />
                            </label>
                            <label>אימות סיסמא
                                <input 
                                    type="password"
                                    onChange={this.onConfirmPasswordChange}
                                    onBlur={this.checkCompatibilityPasswords}
                                    required
                                />
                            </label>
                            <label>תמונת פרופיל
                                <input type="file" onChange={this.onProfileChange}/>
                            </label>
                            {this.state.profile && <img src={this.state.profile}/>}
                            <button className="signup__box">צור חשבון</button>
                        </form>
                    </div>
                </div>
            </div>
        );
    };
};

const mapPropsToDispatch = (dispatch) => ({
    startUploadImage: (picture) => dispatch(startUploadImage(picture)),
    startCreateNewUser: (user) => dispatch(startCreateNewUser(user))
});


export default connect(undefined,mapPropsToDispatch)(CreateAccountForm);