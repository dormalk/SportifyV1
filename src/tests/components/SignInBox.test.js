import React from 'react';
import { shallow } from 'enzyme';
import { SignInBox } from '../../components/SignInBox';

test('should render SignInBox correctly', () => {
    const wrapper = shallow(<SignInBox startLoginOnGoogle={() => { }}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should call startLoginOnGoogle on button click', () => {
    const startLoginOnGoogle = jest.fn();
    const wrapper = shallow(<SignInBox startLoginOnGoogle={startLoginOnGoogle}/>);
    wrapper.find('.login__google').simulate('click');
    expect(startLoginOnGoogle).toHaveBeenCalled();
});


test('should set email on input change', () => {
    const value = 'example@gmail.com';
    const wrapper = shallow(<SignInBox />);
    wrapper.find('input').at(0).simulate('change', {
      target: { value }
    });
    expect(wrapper.state('email')).toBe(value);
});


test('should set password on input change', () => {
    const value = '12345';
    const wrapper = shallow(<SignInBox />);
    wrapper.find('input').at(1).simulate('change', {
      target: { value }
    });
    expect(wrapper.state('password')).toBe(value); 
});


test('should call startLoginOnEmail on submit form if email and password valid', () => {
    const startLoginOnEmail = jest.fn();
    const password = '123456';
    const email = 'example@gmail.com';

    const wrapper = shallow(<SignInBox startLoginOnEmail={startLoginOnEmail} password={password} email={email}/>);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => { }
    });
    expect(startLoginOnEmail).toHaveBeenCalled();
});


test('should not call startLoginOnEmail on submit form if password not valid', () => {
    const startLoginOnEmail = jest.fn();
    const password = '1234';
    const email = 'example@gmail.com';

    const wrapper = shallow(<SignInBox startLoginOnEmail={startLoginOnEmail} password={password} email={email}/>);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => { }
    });
    expect(startLoginOnEmail).not.toHaveBeenCalled();
});


test('should not call startLoginOnEmail on submit form if email not valid', () => {
    const startLoginOnEmail = jest.fn();
    const password = '123456';
    const email = 'examplegmail.com';

    const wrapper = shallow(<SignInBox startLoginOnEmail={startLoginOnEmail} password={password} email={email}/>);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => { }
    });
    expect(startLoginOnEmail).not.toHaveBeenCalled();
});