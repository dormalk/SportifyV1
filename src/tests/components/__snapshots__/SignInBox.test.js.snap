// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignInBox correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="login-box"
  >
    <h1
      className="login-box__title"
    >
      Login Sportify
    </h1>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <input
        autoFocus={true}
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <div
        className="login__non-provider"
      >
        <button
          className="login_email"
          type="submit"
        >
          SIGN IN
        </button>
        <Link
          className="signup"
          replace={false}
          to="/signup"
        >
          CREATE ACCOUNT
        </Link>
      </div>
      <button
        className="login__google"
        onClick={[Function]}
        type="button"
      >
        SIGN IN WITH GOOGLE
      </button>
    </form>
  </div>
</div>
`;
